care/facility/models/file_upload.py

```diff
@@ -1,79 +1,53 @@
-import enum
 import time
 from uuid import uuid4
 
 import boto3
+from django.contrib.auth import get_user_model
 from django.db import models
 
-from care.facility.models import FacilityBaseModel
-from care.users.models import User
 from care.utils.csp.config import BucketType, get_client_config
+from care.utils.models.base import BaseManager
 
+User = get_user_model()
 
-class FileUpload(FacilityBaseModel):
-    """
-    Stores data about all file uploads
-    the file can belong to any type ie Patient , Consultation , Daily Round and so on ...
-    the file will be uploaded to the corresponding folders
-    the file name will be randomised and converted into an internal name before storing in S3
-    all data will be private and file access will be given on a NEED TO BASIS ONLY
-    """
-
-    # TODO : Periodic tasks that removes files that were never uploaded
-
-    class FileType(enum.Enum):
-        PATIENT = 1
-        CONSULTATION = 2
-        SAMPLE_MANAGEMENT = 3
-        CLAIM = 4
-        DISCHARGE_SUMMARY = 5
-        COMMUNICATION = 6
-        CONSENT_RECORD = 7
 
-    class FileCategory(enum.Enum):
+class BaseFileUpload(models.Model):
+    class FileCategory(models.TextChoices):
         UNSPECIFIED = "UNSPECIFIED"
         XRAY = "XRAY"
         AUDIO = "AUDIO"
         IDENTITY_PROOF = "IDENTITY_PROOF"
 
-    FileTypeChoices = [(e.value, e.name) for e in FileType]
-    FileCategoryChoices = [(e.value, e.name) for e in FileCategory]
+    external_id = models.UUIDField(default=uuid4, unique=True, db_index=True)
 
     name = models.CharField(max_length=2000)  # name should not contain file extension
     internal_name = models.CharField(
         max_length=2000
     )  # internal_name should include file extension
     associating_id = models.CharField(max_length=100, blank=False, null=False)
-    upload_completed = models.BooleanField(default=False)
-    is_archived = models.BooleanField(default=False)
-    archive_reason = models.TextField(blank=True)
-    uploaded_by = models.ForeignKey(
-        User,
-        on_delete=models.PROTECT,
-        null=True,
-        blank=True,
-        related_name="uploaded_by",
-    )
-    archived_by = models.ForeignKey(
-        User,
-        on_delete=models.PROTECT,
-        null=True,
-        blank=True,
-        related_name="archived_by",
-    )
-    archived_datetime = models.DateTimeField(blank=True, null=True)
-    file_type = models.IntegerField(
-        choices=FileTypeChoices, default=FileType.PATIENT.value
-    )
+    file_type = models.IntegerField(default=0)
     file_category = models.CharField(
-        choices=FileCategoryChoices,
-        default=FileCategory.UNSPECIFIED.value,
+        choices=FileCategory.choices,
+        default=FileCategory.UNSPECIFIED,
         max_length=100,
     )
+    created_date = models.DateTimeField(
```

Comment:
inherit from facility base model.

Reply:
inhereting the basefileupload from base modle might add the new user dependent fields https://github.com/coronasafe/care/issues/1697
that's why I avoided inheriting from it

Reply:
Okays